network/test/middleware_test.go

```diff
@@ -4,16 +4,19 @@ import (
 	"context"
 	"fmt"
 	"sync"
+	"sync/atomic"
```

Comment:
Can you please run your new tests with `-count 100 -failfast -race` to make sure they won't flake?

Reply:
Passing about 80-90% of the time. The failure runs that happen are because network is not processing messages fast enough, which is more than likely due to running the test back to back (spawning new nodes, allocating ports etc).